src/checks/software_selection.ts

```diff
@@ -15,3 +15,19 @@ export function selectSinglePattern(pattern: string) {
     await softwareSelection.close();
   });
 }
+
+export function selectPatterns(patterns: string[]) {
+  it(`should select patterns ${patterns.join(", ")}`, async function () {
+    const sidebar = new SidebarPage(page);
+    const software = new SoftwarePage(page);
+    const softwareSelection = new SoftwareSelectionPage(page);
+
+    await sidebar.goToSoftware();
+    await software.changeSelection();
+
+    for (const pattern of patterns) {
+      await softwareSelection.selectPattern(pattern);
+    }
```

Comment:
For asynchronous programming make sense to use this instead of loops (sorry I suggest you to delete in some point, I don't remember, probably would look to me like auto-generated code without explanation).

`await Promise.all(patterns.map(pattern => softwareSelection.selectPattern(pattern)));`

Reply:
Changed.

Reply:
> For asynchronous programming make sense to use this instead of loops (sorry if I suggest you to delete some use of this in some point, I don't remember exactly, probably it would had looked to me like auto-generated code without explanation).
> 
> `await Promise.all(patterns.map(pattern => softwareSelection.selectPattern(pattern)));`

looks like that we need to do it sequentially, tried with upper loc and [failed](https://openqa.suse.de/tests/17133517#step/agama/8). WDYT?

Reply:
you are right, that would be using parallel programming, we don't want that. Try this, it is pretty much the same but without some characters:
```suggestion
      for (const pattern of patterns) await softwareSelection.selectPattern(pattern);
```

Reply:
changed and [run to verified](https://openqa.suse.de/tests/17159516)